packages/lodestar/src/network/peers/peerManager.ts

```diff
@@ -215,6 +215,19 @@ export class PeerManager {
     this.pingAndStatusTimeouts();
   }
 
+  hasBeaconCommitteeSubnetPeer(subnet: number): boolean {
```

Comment:
This function should look directly into gossipsub data, not the peer's metadata. Peers can lie, so what's the ultimate truth is the gossipsub state. @wemeetagain what do you recommend to look at?

Also whatever code is in hasBeaconCommitteeSubnetPeer() has to be extremely fast to not add more performance load to the node. Current implementation I think is way to heavy

Reply:
You can look at `gossipsub.topics`, which is a `Map<topicString, Set<peerIdString>>`.
This is a map of all our subscribed topics to a set of all connected peers that are also subscribed to the topic.

There's also `gossipsub.mesh`, which is also a  `Map<topicString, Set<peerIdString>>`.
This is a map of all our subscribed topics to a set of all connected peers that are also subscribed to the topic and also in the mesh.

The gossipsub publishing behavior is:  We will try to publish to mesh peers in the topic first. If we have no mesh peers, we get some random peers subscribed in the topic (that have a good enough score), and publish to them.

Reply:
since we have `floodPublish` option as `true` and we're the publisher, we'll publish to all topic peers that meet `publishThreshold` score https://github.com/ChainSafe/js-libp2p-gossipsub/blob/349d5a6303b9f1cd6e73819a495c1d42609bcb3f/ts/index.ts#L1103

the issue is we don't know which topic peers meet that condition unless we duplicate the condition in lodestar. The best way is to return number of `toSend` peers in `libp2p-interface` & `libp2p-gossipsub` in the `publish()/_publish()` method so that we can use it in lodestar metric, is that ok @wemeetagain ?

Reply:
@dapplion I created https://github.com/libp2p/js-libp2p-interfaces/pull/119